v0/destinations/zendesk/transform.js

```diff
@@ -45,6 +46,86 @@ function responseBuilder(message, headers, payload, endpoint) {
   return response;
 }
 
+/**
+ * Returns the payload for updating primary email of users.
+ * @param {*} userIdentityId -> userIdentity Id
+ * @param {*} userId -> userId of users
+ * @param {*} headers -> Authorizations for API's call
+ * @returns
+ */
+const responseBuilderToSetPrimaryAccount = (
+  userIdentityId,
+  userId,
+  headers
+) => {
+  const response = defaultRequestConfig();
+  const updatedHeaders = {
+    ...headers,
+    "X-Zendesk-Marketplace-Name": ZENDESK_MARKET_PLACE_NAME,
+    "X-Zendesk-Marketplace-Organization-Id": ZENDESK_MARKET_PLACE_ORG_ID,
+    "X-Zendesk-Marketplace-App-Id": ZENDESK_MARKET_PLACE_APP_ID
+  };
+  response.endpoint = `${endPoint}users/${userId}/identities/${userIdentityId}/make_primary`;
+  response.method = defaultPutRequestConfig.requestMethod;
+  response.headers = updatedHeaders;
+  return response;
+};
+
+/**
+ * ref: https://developer.zendesk.com/api-reference/ticketing/users/user_identities/#list-identities
+ * ref: https://developer.zendesk.com/api-reference/ticketing/users/user_identities/#create-identity
+ * This function search identity from userId and fetch its details, if that identity doesn't exists
+ * then it will create the identity and takes the id for it.
+ * @param {*} userId -> userId of users
+ * @param {*} primaryEmail -> primary email passed in payload used to set this email as primary account.
+ * @param {*} headers -> Authorizations for API's call
+ * @returns it returns identityId or undefined if it gives error.
+ */
+const getUserIdentityId = async (
+  userId,
+  primaryEmail,
+  headers,
+  destinationConfig
+) => {
+  const url = `${endPoint}users/${userId}/identities`;
+  const config = { headers };
+  try {
+    const resp = await axios.get(url, config);
+    if (!resp || !resp.data || resp.data.count === 0) {
```

Comment:
If the count=0 we don't need to through error we can create the user.

Reply:
count === 0 means that there is no identity of the user, but in our flow for setting the email as primary, there must be 1 email of the user which means the count is always greater than zero, otherwise it's unable to fetch the user identities.

Reply:
Yes, we can create the user and add the email to that user.

Reply:
Okay, got it, will update.

Reply:
Actually, this case will never come, this is just for one more API response validation, As for updating the primary email, we need userId and we are taking userId only from the email, so that means the email should always be there and the count is greater than 0.

Reply:
What if the customer provides `primaryEmail` with the first API call?

Reply:
updated